### PR TITLE
Remove util module and use securesystemslib key interfaces instead

### PR DESCRIPTION
**Pull request by lukpueh**
NOTE: Requires secure-systems-lab/securesystemslib#288 to be merged and released. Currently the required securesystemslib are pulled in via f1f02f4, which should be removed before merging here.

**Fixes issue #80**:

**Description of the changes being introduced by the pull request**:

With https://github.com/secure-systems-lab/securesystemslib/pull/288 in place we no longer need to wrap or implement key handling (generation + import) functions in an 'in_toto.util' module. This PR:
- replaces calls to 'in_toto.util' with calls to 'securesystemslib.interface' and 'securesystemslib.gpg.functions'
- adds sphinx docs for securesystemslib functions that might be useful for in-toto users
  (see demo on [in-toto-lukpueh.readthedocs.io](https://in-toto-lukpueh.readthedocs.io/en/latest/api.html#key-utilities))
- removes 'in_toto.util' and tests
- adds command line flags to 'in-toto-run', 'in-toto-record' and 'in-toto-sign' to take and/or prompt for a signing key decryption password
- cleans up tests (adds a test key generation mixin for DRY test code, removes obsolete test code)

As usual, details can be found in the commit messages.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Review by trishankatdatadog — APPROVED** [review]
Not sure if my vote counts, but I don't see anything I don't like, great job!

**Review by adityasaky — APPROVED** [review]
This looks good to me, @lukpueh, thanks! Excited for 1.0 :smile:

File: requirements-pinned.txt [hunk 1]
```diff
@@ -35,6 +35,6 @@ pathspec==0.8.0
 pycparser==2.20           # via cffi
 pynacl==1.4.0             # via securesystemslib
 python-dateutil==2.8.1
-securesystemslib[crypto,pynacl]==0.16.0
+git+https://github.com/lukpueh/securesystemslib@misc-interface-enhancements
```
Comment (SantiagoTorres) [inline requirements-pinned.txt:38]: we should just be sure to update this whenever releasing. I wonder what's the best way to signal this

**Review by SantiagoTorres — APPROVED** [review]
Fantastic work!

File: requirements-pinned.txt (hunk 1 above)
Comment (lukpueh) [reply requirements-pinned.txt:38]: Oh, I probably would have waited until secure-systems-lab/securesystemslib#288 was merged. But I'm sure I won't forget to remove the temporary pinning commit f1f02f4 before we release 1.0.0.

**Pull request merged**